### ISSUE TITLE
Files are not properly ignored from .gitignore

### ISSUE THREAD

**Issue by vesk4000** [XAMPPRocky/tokei#877]
In my `.gitignore` file I have the line `*.sql`. All that does is it ignores the SQL files in the root directory of my repository. Git works perfectly with that. But tokei seems to ignore all of my `.sql` files no matter what directory they are in. If I wanted to do that I would've written `**/*.sql`. So yeah pretty sure that's a bug, and it would be pretty cool if it was fixed.

**Comment by LovecraftianHorror** [comment]
From https://git-scm.com/docs/gitignore:

> - If there is a separator at the beginning or middle (or both) of the pattern, then the pattern is relative to the directory level of the particular .gitignore file itself. Otherwise the pattern may also match at any level below the .gitignore level.

So `tokei` is correct here to ignore all SQL files since the match can occur at any level below where you `.gitignore` is found. 

Beyond that I just tried with a fresh git repo and wasn't able to reproduce. `*.sql` made `git` ignore all SQL files while `/*.sql` made `git` only ignored SQL files in the root. Is it possible that you forcibly added them at some point, or that they were added before the `.gitignore` entry?

**Comment by vesk4000** [comment]
@LovecraftianHorror Yup, you are completely right. I'm sorry for the trouble. I'm kind of still a beginner at Git. I thought I knew how the globbing patterns that Git uses work, but apparently not quite. And I hadn't even realized that my `.gitignore` wasn't even working as I intended. No bug on tokei's part then. I suppose it's a good thing that I tried to add the tokei badge, cause otherwise I wouldn't even know that my `.gitignore` was wrong. Thanks for the help!
